multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/helpers/MtaDescriptorPropertiesResolver.java

```diff
@@ -118,6 +113,24 @@ private DeploymentDescriptor correctEntityNames(DeploymentDescriptor descriptor)
                       .validate();
     }
 
+    private void replaceNullLiveRoutesWithReferenceToRoute(DeploymentDescriptor descriptor) {
+        for (Module module : descriptor.getModules()) {
+            Map<String, Object> moduleParameters = module.getParameters();
+
+            List<Map<String, Object>> routes = RoutesValidator.applyRoutesType(moduleParameters.get(SupportedParameters.ROUTES));
+
+            for (Map<String, Object> routeMap : routes) {
+                Object liveRoute = routeMap.get(SupportedParameters.LIVE_ROUTE);
+
+                if (Objects.isNull(liveRoute)) {
+                    int routeIndex = routes.indexOf(routeMap);
+
+                    routeMap.put(SupportedParameters.LIVE_ROUTE, String.format(ROUTE_REFERENCE_PLACEHOLDER, routeIndex));
```

Comment:
where is the validation that the route will not be overridden from the user

Reply:
in ReadOnlyParametersChecker. if the user tries to overwrite it the build will fail